File: BSharp/Controllers/DTO/GetResponse.cs
using System.Collections;
using System.Collections.Generic;

namespace BSharp.Controllers.DTO
{
    /// <summary>
    /// Represents a response of raw entities for save + some search metadata
    /// </summary>
    public class GetResponse<TDto> : EntitiesResponse<TDto> where TDto : DtoBase
    {
        public int Skip { get; set; }

        public int Top { get; set; }

        public string OrderBy { get; set; }

        public int TotalCount { get; set; }
    }
}

File: BSharp/ClientApp/src/app/data/dto/get-response.ts
import { EntityWithKey } from '../entities/base/entity-with-key';
import { Entity } from '../entities/base/entity';

export class EntitiesResponse<TDto extends Entity = EntityWithKey> {
  Bag: { [key: string]: any; };
  Result: TDto[];
  CollectionName: string;
  RelatedEntities: { [key: string]: EntityWithKey[]; };
}

export class GetResponse extends EntitiesResponse {
  Skip: number;
  Top: number;
  OrderBy: string;
  Desc: boolean;
  TotalCount: number;
}
